interfaces/builtin/dsp.go

```diff
@@ -0,0 +1,107 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/udev"
+)
+
+const dspSummary = `allows controlling digital signal processors on certain boards`
+
+const dspBaseDeclarationSlots = `
+  dsp:
+    allow-installation:
+      slot-snap-type:
+        - gadget
+        - core
+    deny-auto-connection: true
+`
+
+const ambarellaDspConnectedPlugApparmor = `
+# Description: can manage and control the integrated digital signal processor on
+# the ambarella device. This allows privileged access to hardware and kernel 
+# drivers related to the digital signal processor and thus is super-privileged.
```

Comment:
This comment  claims the interface is super-privileged but there is nothing in the base declaration which makes it such that I can see - should there be a

```go
const dspBaseDeclarationPlugs = `
  dsp:
    allow-installation: false
    deny-auto-connection: true
`
```

Reply:
You're right, this interface was refactored to be provided by a gadget instead of snapd, and the interface is no longer super-privileged. I will adjust the comment here to reflect that. You can think of this interface as similar to gpio or pwm, etc. but technically it could be more powerful, allowing direct access to a co-processor as opposed to just hardware state of pins, but in some deployments access to the gpio or pwm pins could be considered more privileged than the DSP, so it's dependent on how it's used. Hence why this slot is now provided by a gadget.

Reply:
Thanks - +1 from me.